specs/AdaptiveCardsXamarinSpec.md

```diff
@@ -0,0 +1,188 @@
+| Custom Action | ```(ACRBaseActionElementRenderer*) getActionRenderer: (NSNumber*)``` | P1 |
+| Custom Action Set | ```(void) setActionSetRenderer: (id<ACRIBaseActionSetRenderer>)``` | P2 |
+| Custom Action Set | ```(id<ACRIBaseActionSetRenderer>) getActionSetRenderer``` | P2 | 
+| Resource Resolvers | ```(void) setResourceResolver: (NSObject<ACOIResourceResolver>) scheme: (NSString*)``` | P2 |
+| Resource Resolvers | ```(NSObject<ACOIResourceResolver>) getResourceResolverForScheme: (NSString*)``` | P2 |
+| Feature Registration | ```(void) addFeature: (NSString*) featureVersion: (NSString*)``` | P0 | 
+| Feature Registration | ```(NSString*) removeFeature: (NSString*)``` | P0 |
+| Feature Registration | ```(NSString*) getFeatureVersion: (NSString*)``` | P0 |
+| Resource Information | ```(NSArray<ACORemoteResourceInformation>*) remoteResourceInformation``` | P2 | 
+
+## Breaking changes
+
+There is one big breaking change to be introduced in the iOS platform due to the Xamarin bindings having a direct relationship with the name of the library, which requires changing the library name from ACFramework to AdaptiveCards.
+
+Besides that, no breaking changes should be introduced as the native platforms' code is only consumed and projected into C#.
+
+## Test Plan
+
+For the alpha and pre-release versions the testing will be performed manually
+
+## Release plan
+
+### Artifacts
+
+The artifacts to be released with this library will be:
+1. Source code for Xamarin bindings in the github repository
+1. Xamarin.Android nuget package
+1. Xamarin.iOS nuget package
+
+### Continuous Integration and Release
+
+To be able to release the previously mentioned nuget packages build definitions must be created. The current system we use for building the other packages allow us to build Xamarin.Android and Xamarin.iOS packages
+
+#### Pre-requisites
+
+The Xamarin packages require native libraries to be built beforehand, as we can't upload binaries into the repository, then the Android and iOS build definitions must be updated to copy the generated binaries into the xamarin projects folder. 
+
+#### Build definitions
+
+At least four build pipelines will be needed for this project, 2 pipelines for Continuous integration and 2 pipelines for the release. The pipelines for the release will be similar to the UWP and WPF pipelines that generate nuget packages and build against a visual studio solution/project.
+
+## Remaining Tasks
+
+For the alpha version all P0 and P1 APIs should be done while P2 and P3 tasks must be done for the first complete release. 
+
+## Open issues
+
+Currently the naming convention for AdaptiveCards packages are medium sized, as Xamarin builds on top of other platforms, these namings internally become large enough to overpass the character limit so errors tend to appear during development which make it harder (Xamarin also doesn't spew enough information to be easily solved).
```

Comment:
[Team Spec Review]  Understand how this will impact the reliability of both (1) Builds ( CI + Dev Inner loop) as well as (2) Consumption of the released packages; For 1, check with Paul on whether CI/ADO provides affordances on enabling long file name enabling on teh Windows machines; If not, an alternative could be to compile the Xamarin packages on Linux machines in our packageEs pool.